### PR TITLE
LPS-108803 Disabling modules preloading does not work

### PR DESCRIPTION
**Pull request by markocikos**
Hey @wincent ,

A followup on https://github.com/wincent/liferay-portal/pull/168

Changes:
- Preload option is now showing in UI. As we cannot configure hiding of just one option, I deleted the option to completely turn off AUI.

Please review the code.

Thank you!

/cc @izaera @jbalsas

### CODE REVIEW

**Comment by wincent** [comment]
Thanks @markocikos.

Pretty much what I said [over here](https://github.com/wincent/liferay-portal/pull/168#issuecomment-596472869):

> I think that we should probably hold off on sending it to Brian because we're still trying to stabilize and cut the GA2 release, and there may be some subtle bugs that we'll expose if we make these changes.
>
> In terms of actually reviewing the code, your explanation makes sense to me

I did some "QA" with preloading off (clicking around guest and non-guest pages, reloading etc) and couldn't spot any errors.

Before (with preloading):

preload

After (without preloading):

no-preload

So, it drops from 385 to 363 requests, and 7.1 MB to 6.8 MB (those numbers in `NODE_ENV=development`, would expect them to be different in production).

**Comment by wincent** [comment]
Forgot to mention; probably need a `gradle buildLang` to update the language properties files, right?

**Comment by jbalsas** [comment]
> So, it drops from 385 to 363 requests, and 7.1 MB to 6.8 MB (those numbers in NODE_ENV=development, would expect them to be different in production).

That's interesting... if I counted correctly, we have 39 AUI_PRELOAD modules. Based on your numbers I think that means that we were preloading 22 modules that aren't used off the bat. It also means that we're now not preloading 17 that are requested a bit later on... which is probably making it a bit worse as seen per the .4s additional load time (I know this is not a reliable/scientific measurement).

I'm wondering if it would make sense to identify those additional 17 modules and move them to "core" as well and then work our way from there... 🤔

**Comment by wincent** [comment]
> which is probably making it a bit worse as seen per the .4s additional load time (I know this is not a reliable/scientific measurement).

In a couple of runs I saw *huge* variability in run times (emptying cache and hard-reloading each time), so yeah, we can't really draw any solid conclusions. The other complicating factor is this was just on a random guest page run (with the hello world widget, site nav, and the usual other ones), and I'd expect the numbers to vary based on what you have on the page too.

Given that this is gated by configuration (which defaults to preloading on — ie. no change), and we can't know what is going to be on client pages, I don't know how much we can expect to get from fine tuning the list. The real benefit for most clients isn't going to come until we turn AUI off by default, which is still some way off in the future.

**Comment by jbalsas** [comment]
> Given that this is gated by configuration (which defaults to preloading on — ie. no change), and we can't know what is going to be on client pages, I don't know how much we can expect to get from fine tuning the list.

Fair enough

> The real benefit for most clients isn't going to come until we turn AUI off by default, which is still some way off in the future.

I think the goal would be to see real benefit even before we do so. For example, even if we still load some of AUI core, if we managed to reduce it only to real core (loader), that should still bring a big benefit to our clients.

**Comment by wincent** [comment]
Here are the requests made with preloading on:

https://gist.github.com/wincent/248364c1b7b81910410e771ec3b93240

And with preloading off:

https://gist.github.com/wincent/2779cc464de35816f63c4ab1cbbdaee9

Here is the diff:

https://gist.github.com/wincent/0ff959210966f2ade06d8cc2a11fb985

For reference, made that by exporting a ".har" archive of each and then printing it with:

```js
console.log(require('/Users/greghurrell/Desktop/0.0.0.0.har.json').log.entries.map(e => e.request.url).join('\n'))
```

(Then sorting, and diffing.)

**Comment by jbalsas** [comment]
The difference in URLs is a bit worrisome... why would changing the preload option make the searchQuery change for some of the modules and not others? 🤔

**Comment by wincent** [comment]
Not sure about that. Maybe @izaera knows?

**Comment by jbalsas** [comment]
Maybe that's just what I was wondering before... chances are those are now being requested after the fact through the YUI loader which automatically appends those...

**Comment by izaera** [comment]
Mmm, I think the difference is because: 

1. when you preload: the backend architecture writes the URL to the HTML directly (thus, no params because there's no context to derive the params)
2. when you request from a page: the JS architecture (probably AUI loader) composes a URL based on the current context and appends the params automatically

We do 2 with the AMD loader too, and always append `languageId` whether it is used or not.

**Comment by markocikos** [comment]
> Forgot to mention; probably need a `gradle buildLang` to update the language properties files, right?

Long time ago, I was instructed not to this, that Brian does it. I believe he has auto translation system integrated, so the automatic copies that we generate end up overridden.

**Comment by jbalsas** [comment]
> Long time ago

starwars

We now have an integrated translation system where people will add translations. I think you have to run it now.

**Comment by markocikos** [comment]
> I think the goal would be to see real benefit even before we do so. For example, even if we still load some of AUI core, if we managed to reduce it only to real core (loader), that should still bring a big benefit to our clients.

Please see my performance testing and analysis [here](https://docs.google.com/document/d/11WaMRevHbmU8prYqGAiabrCTexG-G7ot6u6-ECzdAK8/edit).

TLDR: We get no immediate benefit, but future focus on bottlenecks should be more effective.

**Comment by jbalsas** [comment]
LGTM!

Could you please start with the technical Draft for the epic? I think you remember how to do it. Please create the task in the epic and place the draft in the draft folder as usual. I think you can append the performance analysis there.

**Comment by wincent** [comment]
Rebasing this seeing as GA2 is delayed again. If CI is green, will forward.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:stable

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
> 828 out of 1010

This is never, ever going to be green. 😂

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
So, when the CI run doesn't even start, how long do you wait before you give up and try again?

**Comment by john-co** [comment]
ci:stop

**Comment by john-co** [comment]
will resend with reduced test scope.

**Comment by john-co** [comment]
rebased, included relevant test.properties scope, and sent to https://github.com/wincent/liferay-portal/pull/182

**Comment by john-co** [comment]
ci:close

**Comment by john-co** [comment]
ci:open

**Comment by john-co** [comment]
ci:reopen

**Comment by john-co** [comment]
retest this when next [master stable](https://test-1-0.liferay.com/job/test-portal-testsuite-upstream-controller(master_stable)/?auto_refresh=true) passes

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
ci:stop

**Comment by john-co** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Seeing as this one still looks stuck, I am going to rebase it (which I expect will make no difference, but doesn't hurt), and retry forwarding.

**Comment by manoelcyreno** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Just rebased [like I said I would](https://github.com/wincent/liferay-portal/pull/170#issuecomment-604883248), so I am going to stop and re-start the CI.

**Comment by wincent** [comment]
ci:stop

**Comment by wincent** [comment]
ci:forward

**Comment by manoelcyreno** [comment]
ci:stop

**Comment by manoelcyreno** [comment]
ci:test:relevant

**Comment by manoelcyreno** [comment]
Just started reviewing :)

**Comment by wincent** [comment]
Replaced by: https://github.com/wincent/liferay-portal/pull/188
